### ISSUE TITLE
Spark SQL Functions Coverage and Parity - CSV

### ISSUE THREAD

**Issue by shehabgamin** [lakehq/sail#225]
- [x] from_csv
- [ ] schema_of_csv
- [ ] to_csv

**Comment by rafafrdz** [comment]
@shehabgamin ey! Can I get this issue? 😄

**Comment by shehabgamin** [comment]
> [@shehabgamin](https://github.com/shehabgamin) ey! Can I get this issue? 😄

@rafafrdz Done! Thank you so much!!

**Comment by shehabgamin** [comment]
Hey @rafafrdz, just a quick note: Sail has been collaborating closely with DataFusion (cc @alamb, @andygrove) on a shared [datafusion-spark crate](https://github.com/apache/datafusion/tree/main/datafusion/spark), with a current focus on function support.

Would you be open to contributing your work to the `datafusion-spark` crate? My thought is to keep the implementation in both repositories for now, so Sail users can access new functions immediately without having to wait for the next DataFusion release. Once the new DataFusion version is out, we can remove the duplicate code from the Sail codebase.

Development might also be easier on the Sail side first, since we already have a comprehensive test suite for these functions and for testing Spark parity in general. The idea is that the function implementation and tests can then be copied over to DataFusion with minimal friction.

If it’s easier, I’m also happy to handle the upstream contribution myself. I just want to make sure you get full credit for your work. Let me know what you think!

**Comment by rafafrdz** [comment]
Yes sure! I don't mind. In fact, I love the idea of collaborating with both in some way 😃

**Comment by anarefolio** [comment]
Hi, I'd like to work on implementing to_csv, Can I get this issue?
